pkg/reconciler/taskrun/resources/apply.go

```diff
@@ -97,6 +98,15 @@ func ApplyResources(spec *v1beta1.TaskSpec, resolvedResources map[string]v1beta1
 	return ApplyReplacements(spec, replacements, map[string][]string{})
 }
 
+// ApplyPipelineTaskContexts applies the substitution from $(context.pipelineTask.*) with the specified values.
+// Uses "" as a default if a value is not available.
```

Comment:
It looks like `pt.Retries` will be 0 if a value is not provided.  so this will set the variable to `"0"`.  Is that right?

Reply:
Yes, I think zero is the expected value for us. If `retries` of a PipelineTask is set to 5, this means it will run 6 times at most (first run + 5 retries). If a PipelineTask doesn't specify `retries`, it will only run one time. The retry is supposed to be 0(first run + 0 retry) under this circumstance.

Reply:
So I think this code comment might not be quite right - it says `""` is the default if a value is not available but it sounds like `"0"` is the default if a value is not available.